src/blog/2023/011-form-js-1.6.md

```diff
@@ -0,0 +1,98 @@
+---
+
+title: Working with Array Data in form-js@1.6
+description: "Discover the new dynamic list and table components. Learn how they allow you to handle array data with ease."
+preview_image: http://bpmn.io/assets/attachments/blog/2023/011-dynamic-list-readonly.png
+layout: blogpost.hbs
+slug: 2023-form-js-1-6-dynamic-list-and-table
+author:
+  - Vinicius Goulart <https://github.com/vsgoulart>
+  - Valentin Serra <https://github.com/skaiir>
+published: 2023-12-19 12:00
+releases:
+  - 'form-js@1.6.0'
+
+---
+
+<p class="introduction">
+  The release of [form-js@1.6](https://github.com/bpmn-io/form-js) didn't take long, but its impact on handling array data is a game changer. This update introduces two powerful components: tables for elegant tabular data display and dynamic lists for crafting flexible, repeatable, and editable form sections. We designed these components to allow working with complex data arrangements without altering the familiar workflow of the form builder.
+</p>
+
+<!-- continue -->
+
+In our continuous effort over recent months, we wanted to build a powerful way to deal with arrays of data. Our goal was to enable the handling of arrays of any data structures, giving you complete control over your data outputs. We settled on a very flexible solution while maintaining the ease and familiarity of the current form builder interactions.
+
+
+## Dynamic Lists, the Flexible Solution to Array Data
+
+Our new dynamic list component brings flexibility to form-js, enabling users to dynamically add or remove repetitions of a form section while handling all the data mapping for you out of the box.
+
+<div class="figure full-size">
+  <img src="{{ assets }}/attachments/blog/2023/011-dynamic-list.gif">
+  <p class="caption">
+    Add or remove entries of the list with a single click.
+  </p>
+</div>
+
+Customize the behavior of your dynamic list. Decide whether adding and removing items should be possible, how many default items to render, how many items to show uncollapsed, and more. We give you quite a lot of control in configuring this component.
+
+For example, if you want to use dynamic lists as a read-only array data renderer, simply disable the ability to add/remove entries on the dynamic list, make all the child fields read-only, and voilà.
+
+<div class="figure full-size">
+  <img src="{{ assets }}/attachments/blog/2023/011-dynamic-list-readonly.png">
+  <p class="caption">
+    Use the component however you like. It opens up many avenues for creative use cases.
+  </p>
+</div>
+
+The best part is that you still get all the features you expect from simple elements in your dynamic list repetitions. You may now use the `this` variable within FEEL expressions to refer to the local 'scope' of a dynamic list's rendered element.
+
+As a small heads-up, we plan to change how these scopes work, so keep this in mind.
```

Comment:
What do we want to say here? That users shall not use `this`? 

Whatever we ship (and document in a blog) must be backwards compatible _or_ forward-migratable. Let's be clear if this will break in the future, or not mention it at all.

Reply:
I agree with @nikku I would just not mention this at all now until we have committed to the change

Reply:
Actually the `this` variable is backwards compatible. The main change is that we're going to make it implicit in the future. So I guess it's still fine to mention it here.